Chore(npm to yarn): Use yarn as package manager instead of npm, and u…

chykehyman opened the pull request:
###What does this PR do?

Use yarn as package manager instead of npm, and upgrade packages

###Description of Task to be completed?

* Install yarn package manager
* Update each npm package in package.json file using yarn
* Manually install express 4 independent npm packages
* Modify `engine` property of package.json file to latest node and npm versions
* Modify express.js file to reflect proper usage of `express 4` independent npm packages
* Import apiRoute path from `routes.js` file into `express.js` file and use appropriately
* Initialize `express router` in `routes.js` file, and change 'app' to parameter to 'router' variable. Then export the `router` variable
* Find and fix upgrade errors in 404.jade, 500.jade and default.jade files

###How should this be manually tested?

####The following steps would run the application

 1. From your terminal or console window, run `brew install yarn` 
 2. Complete the basic setup of the project gotten from the README, on your local repository.
 3. Cd to the root of the project folder in your console. 
 4. First run the command "yarn install" to install all required and updated node packages including 
     bower components
 5. Run the command `yarn global add gulp-cli`
 6. Create a .env file in root of project folder and add 
     `MONGOHQ_URL=mongodb://localhost:27017/cfh` inside the file
 7. Make sure you have downloaded Mongo DB  and set it up on your PC. Use this => [Download and Setup MongoDB](https://docs.mongodb.com/manual/administration/install-community/)
 8. Create and setup a database using this file [Setup Database](https://docs.mongodb.com/manual/mongo/). Make sure you name the database as `cfh` 
 9. Once `mongod` server is running, run the command `yarn start` or `gulp` to start the application.
 10. Finally, navigate to your browser and enter `localhost:3000`

Any background context you want to provide?

Nil

What are the relevant pivotal tracker stories?

`#152798631`

Screenshots

<img width="733" alt="screen shot 2017-11-29 at 6 21 39 pm" src="https://user-images.githubusercontent.com/22524619/33389205-5fc3631a-d532-11e7-8f88-208595d85b55.png">


<img width="1184" alt="screen shot 2017-11-27 at 2 05 22 pm" src="https://user-images.githubusercontent.com/22524619/33268133-590f0cca-d37c-11e7-84ca-c3eac9b9da03.png">

<img width="1433" alt="screen shot 2017-11-27 at 2 06 37 pm" src="https://user-images.githubusercontent.com/22524619/33268145-64bd3ccc-d37c-11e7-9b40-ace26d9bf48b.png">

chykehyman committed:
Chore(npm to yarn): Use yarn as package manager instead of npm, and upgrade packages

-  Modify src path for gulp sass task and populate common.css file
- Setup yarn package manager
- Update 'engine' property, dependencies and devDependencies package using yarn, in package.json file
- Auto generate yarn.lock file
-  Comment out unnecessary block code(raises  error message on compilation) in 404.jade and 500.jade files
- Modify default.jade file ( change unsupported '!!! 5' to 'doctype html'
- Modify express.js file to reflect 'express 4' independent npm packages
- Import apiRoute path in express.js file and use appropriately
-  Initialize express router in routes.js file,and modify 'app' to 'router' variable

- finishes(#153116431)

daddychukz committed:
feature(jwt-signup-signin): refactor gulp task to compile bootstrap css [Fixes #152798633]

daddychukz committed:
feature(jwt-signup-signin): update coverage badge [Fixes #152798633]

daddychukz committed:
feature(jwt-signup-signin): bootstrap css file [Fixes #152798633]

daddychukz committed:
feature(jwt-signup-signin): generate json web token on user signup/signin [Fixes #152798633]

daddychukz committed:
bug-fix(npm-to-yarn): refactor code indentation [Fixes #153116431]

daddychukz committed:
feature(jwt-signup-signin): set JWT secret key [Finishes #152798633]

daddychukz committed:
chore(npm-to-yarn): resolve merge conflicts [Finishes #153116431]

daddychukz committed:
chore(npm-to-yarn): lock file from yarn [Finishes #153116431]

daddychukz committed:
bug-fix(npm-to-yarn): add signin route [Fixes #153146273]

sachungo commented on app/controllers/index.js:
```diff
@@ -17,7 +17,7 @@ exports.play = function(req, res) {
+  res.render('index', {
+    user: req.user ? JSON.stringify(req.user) : "null"
+  });
 };
```
new line?

sachungo commented on app/views/layouts/default.jade:
```diff
@@ -1,4 +1,4 @@
-!!! 5
+doctype html
```
👍 more clear than the previous syntax

chykehyman replied:
Yea... Thanks

sachungo commented on config/socket/socket.js:
```diff
@@ -16,217 +16,217 @@ module.exports = function(io) {
+    };
 
-};
+};
```
same here - new line

sachungo commented on package.json:
```diff
@@ -1,71 +1,88 @@
+    },
+    "engines": {
+        "node": "8.9.0",
+        "npm": "5.5.1"
```
yarn not npm

sachungo reviewed (commented):
Left minor comments to be addressed. Otherwise, everything looks good

daddychukz committed:
bug(npm-to-yarn): refactor code indentation [Fixes #153146273]

daddychukz committed:
bug(npm-to-yarn): add command to generate bower components [Finishes #153116431]

daddychukz committed:
bug(npm-to-yarn): move gulp packages to dependencies [Finishes #153116431]

daddychukz committed:
bug(npm-to-yarn): move gulp packages to dependencies [Finishes #153116431]

daddychukz committed:
chore(npm-to-yarn): refactor command script [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): add yarn command to procfile [Finishes #153116431]

daddychukz committed:
feature(npm-to-yarn): add yarn command to procfile [Finishes #153116431]

daddychukz committed:
feature(npm-to-yarn): add yarn command to procfile [Finishes #153116431]

daddychukz committed:
feature(npm-to-yarn): add yarn comman [Finishes #153116431]

daddychukz committed:
feature(npm-to-yarn): refactor yarn start script [Finishes #153116431]

daddychukz committed:
feature(npm-to-yarn): refactor yarn start script [Finishes #153116431]

daddychukz committed:
feature(edit-profile): add gulp command to procfile [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): move browser-sync package to dependency [Finishes #153146273]

daddychukz committed:
feature(edit-profile): add gulp command to procfile [Finishes #153146273]

daddychukz committed:
feature(edit-profile): add gulp command to procfile [Finishes #153146273]

daddychukz committed:
feature(edit-profile): add gulp command to procfile [Finishes #153146273]

daddychukz committed:
feature(edit-profile): add gulp command to procfile [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): add start command in procfile [Finishes #153116431]

daddychukz committed:
feature(npm-to-yarn): update package, move to dependency [Finishes #153146273]

daddychukz committed:
feature(npm -to-yarn): add script to run bower on heroku [Finishes #153146273]

daddychukz committed:
bug(npm-to-yarn): refactor start script [Finishes #153146273]

daddychukz committed:
feature(npm-to yarn): refactor procfile [Finishes #15311643]

daddychukz committed:
feature(npm-to yarn): refactor procfile [Finishes #15311643]

daddychukz committed:
feature(npm-to-yarn): refactor start command [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): refactor server.js port [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): remove engines object [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): revert changes [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): refactor procfile [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): refactor start script [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): refactor scripts [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): refactor procfile [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): refactor scripts command [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): refactor scripts [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): refactor scripts command [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): refactor scripts command [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): refactor procfile [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): refactor procfile [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): refactor scripts command [Finishes #153146273]

daddychukz committed:
feature(npm-to-yarn): refactor procfile [Finishes #153146273]

sachungo commented on Procfile:
```diff
@@ -1 +1 @@
-web: ./node_modules/.bin/forever -m 5 server.js
+web: ./node_modules/.bin/forever -m 5 server.js
```
Add new line at end of file

sachungo commented on app/controllers/users.js:
```diff
@@ -132,21 +194,21 @@ exports.addDonation = function(req, res) {
-        }
-      });
+          if (!duplicate) {
+            console.log('Validated donation');
```
Console statement shouldn' be deployed to production code unless it's something important that can't be confirmed using other means i.e. API response, data etc.
You can  add a TODO comment to the console statements, i.e.
```j
//Todo: to be removed
console.log('my statement')

sachungo commented on app/controllers/users.js:
```diff
@@ -1,10 +1,13 @@
 /**
  * Module dependencies.
  */
+var jwt = require('jsonwebtoken');
```
You did merge the JWT PR in here

sachungo commented on public/js/controllers/index.js:
```diff
@@ -15,10 +15,55 @@ angular.module('mean.system')
+        }, (response) => {
+          $scope.alert = response.data.message;
+        });
+    }
```
Has to be addressed

sachungo reviewed (changes requested):
Remove `yarn-error.log` from the git index by executing `git rm --cache <file>` once it's been added to `.gitignore`

Haven't tested locally as I still have to set my dev environment. 

The `yarn add` in the PR description should be `yarn install` as we are not adding new packages but installing those in `package.json`

chykehyman committed:
Chore(npm-to-yarn): fix PR issues

- Add 'yarn-error.log' file to .gitignore file
- Delete cached 'yarn-error.log' file from project index
- Add new line in 'Procfile'
- Remove unnecessary commented code from 'socket.js' file in 'config/socket' folder

sachungo reviewed (approved):
Tested locally and it works
